### ISSUE TITLE
Datatables warning on Dashboard and Jobs views - since update to 1.6.0

### ISSUE THREAD

**Issue by nisori** [bacularis/bacularis-app#7]
Hi,
My installation was updated to 1.6.0 this morning and since then, I get two modal message boxes each time I enter the Dashboard or the Jobs view.

Dashboard:
> DataTables warning: table id=last_jobs_list - Requested unknown parameter 'filetable' for row 0, column 29. For more information about this error, please see http://datatables.net/tn/4

Jobs:
> DataTables warning: table id=job_list - Requested unknown parameter 'filetable' for row 0, column 30. For more information about this error, please see http://datatables.net/tn/4

The column "File Table" appears to be a hidden column, when I select it, the column is empty (as the warnings suggest).

**Comment by nisori** [comment]
Bacula version: 9.6.7
Bacula database version: 16

```
MariaDB [bacula]> explain File;
+------------+----------------------+------+-----+---------+----------------+
| Field      | Type                 | Null | Key | Default | Extra          |
+------------+----------------------+------+-----+---------+----------------+
| FileId     | bigint(20) unsigned  | NO   | PRI | NULL    | auto_increment |
| FileIndex  | int(11)              | YES  |     | 0       |                |
| JobId      | int(10) unsigned     | NO   | MUL | NULL    |                |
| PathId     | int(10) unsigned     | NO   |     | NULL    |                |
| FilenameId | int(10) unsigned     | NO   |     | NULL    |                |
| DeltaSeq   | smallint(5) unsigned | YES  |     | 0       |                |
| MarkId     | int(10) unsigned     | YES  |     | 0       |                |
| LStat      | tinyblob             | NO   |     | NULL    |                |
| MD5        | tinyblob             | YES  |     | NULL    |                |
+------------+----------------------+------+-----+---------+----------------+
9 rows in set (0.002 sec)

MariaDB [bacula]> explain Job;
+-----------------+---------------------+------+-----+---------+----------------+
| Field           | Type                | Null | Key | Default | Extra          |
+-----------------+---------------------+------+-----+---------+----------------+
| JobId           | int(10) unsigned    | NO   | PRI | NULL    | auto_increment |
| Job             | tinyblob            | NO   |     | NULL    |                |
| Name            | tinyblob            | NO   | MUL | NULL    |                |
| Type            | binary(1)           | NO   |     | NULL    |                |
| Level           | binary(1)           | NO   |     | NULL    |                |
| ClientId        | int(11)             | YES  |     | 0       |                |
| JobStatus       | binary(1)           | NO   |     | NULL    |                |
| SchedTime       | datetime            | YES  |     | NULL    |                |
| StartTime       | datetime            | YES  |     | NULL    |                |
| EndTime         | datetime            | YES  |     | NULL    |                |
| RealEndTime     | datetime            | YES  |     | NULL    |                |
| JobTDate        | bigint(20) unsigned | YES  | MUL | 0       |                |
| VolSessionId    | int(10) unsigned    | YES  |     | 0       |                |
| VolSessionTime  | int(10) unsigned    | YES  |     | 0       |                |
| JobFiles        | int(10) unsigned    | YES  |     | 0       |                |
| JobBytes        | bigint(20) unsigned | YES  |     | 0       |                |
| ReadBytes       | bigint(20) unsigned | YES  |     | 0       |                |
| JobErrors       | int(10) unsigned    | YES  |     | 0       |                |
| JobMissingFiles | int(10) unsigned    | YES  |     | 0       |                |
| PoolId          | int(10) unsigned    | YES  |     | 0       |                |
| FileSetId       | int(10) unsigned    | YES  |     | 0       |                |
| PriorJobId      | int(10) unsigned    | YES  |     | 0       |                |
| PurgedFiles     | tinyint(4)          | YES  |     | 0       |                |
| HasBase         | tinyint(4)          | YES  |     | 0       |                |
| HasCache        | tinyint(4)          | YES  |     | 0       |                |
| Reviewed        | tinyint(4)          | YES  |     | 0       |                |
| Comment         | blob                | YES  |     | NULL    |                |
+-----------------+---------------------+------+-----+---------+----------------+
27 rows in set (0.002 sec)
```
Browser: Opera version 95.0.4635.84 (Chromium version:109.0.5414.120)

Let me know what else you need.

**Comment by ganiuszka** [comment]
Hello nisori,

Thank you for reporting this problem.

This problem happens in the Bacularis however in real it is Bacula catalog format issue. Let me describe it in few words.

In version ``1.6.0`` we changed a way of getting data from the catalog database. Previously it was through ORM that had static table column definition. Now the table columns are taken from the catalog without static mapping. This move exposed a Bacula problem that in some MySQL catalog databases the ``FileTable`` column is missing in the ``Job`` table. I researched it deeper and at the moment I reported a bug to the Bacula bug tracker:

https://gitlab.bacula.org/bacula-community-edition/bacula-community/-/issues/2684

To solve this problem you can run the following SQL commands in your MySQL catalog:

```
ALTER TABLE Job ADD COLUMN FileTable CHAR(20) default 'File';
ALTER TABLE JobHisto ADD COLUMN FileTable CHAR(20) default 'File';
```

Alternatively you can wait on a new Bacula version until the same fix will come to a new ``update_mysql_tables`` script.

In next Bacularis version to not show this warning windows we will remove using the ``FileTable`` column from the job tables as it is very internal column. Additionally this column is not used by the Bacula Community yet.

Thanks again.

Best regards,
Marcin Haba (gani)

**Comment by ganiuszka** [comment]
Hello nisori,

I added the default value to ``filetable`` job property. So now even if the ``filetable`` is missing, the error will not appear.

This change will be available from next ``1.7.0`` release.

Thanks.

Best regards,
Marcin Haba (gani)
